### PR TITLE
Add Sign Up Feature

### PR DESCRIPTION
**Pull request by Lordapa**

### CODE REVIEW

File: src/DonateTo.IdentityServer/Controllers/AccountController.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ public IActionResult Register()
 
         [HttpPost]
         [ValidateAntiForgeryToken]
-        public async Task<IActionResult> Register(UserRegistrationModel userModel)
+        public async Task<IActionResult> Register(UserRegistrationViewModel userModel)
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Controllers/AccountController.cs:145]: change the parameter name to match with the type (UserRegistrationViewModel)

File: src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs [hunk 2]
```diff
@@ -2,28 +2,31 @@
 
 namespace DonateTo.IdentityServer.Models
 {
-    public class UserRegistrationModel
+    public class UserRegistrationViewModel
     {
+        [Required(ErrorMessage = "FirstName is required")]
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs:7]: Change the error message to "First name is required"

File: src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs [hunk 3]
```diff
@@ -2,28 +2,31 @@
...
         public string FirstName { get; set; }
 
+        [Required(ErrorMessage = "LastName is required")]
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs:10]: Change the error to "Last name is required"

File: src/DonateTo.IdentityServer/Views/Account/Register.cshtml [hunk 4]
```diff
@@ -1,45 +1,42 @@
...
-                <span asp-validation-for="ConfirmPassword" class="text-danger"></span>
-            </div>
-            <div class="form-group">
-                <input type="submit" value="Create" class="btn btn-primary" />
-            </div>
-        </form>
+<form class="form-card" asp-action="Register">
+    <h3 class="mb-3 font-weight-normal card-title">Sign Up</h3>
+
+    <label asp-for="FirstName" class="sr-only">LastName</label>
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Views/Account/Register.cshtml:10]: Last Name

File: src/DonateTo.IdentityServer/Views/Account/Register.cshtml [hunk 5]
```diff
@@ -1,45 +1,42 @@
...
+    <input type="text" asp-for="FirstName" class="form-control" placeholder="FirstName">
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Views/Account/Register.cshtml:11]: placeholder "First Name"

File: src/DonateTo.IdentityServer/Views/Account/Register.cshtml [hunk 6]
```diff
@@ -1,45 +1,42 @@
...
+    <span asp-validation-for="FirstName"></span>
+
+    <label asp-for="LastName" class="sr-only">LastName</label>
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Views/Account/Register.cshtml:14]: Change to Last Name

File: src/DonateTo.IdentityServer/Views/Account/Register.cshtml [hunk 7]
```diff
@@ -1,45 +1,42 @@
...
+    <input type="text" asp-for="LastName" class="form-control" placeholder="LastName">
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Views/Account/Register.cshtml:15]: Change placeholder to "Last Name"

File: src/DonateTo.IdentityServer/wwwroot/js/site.js [hunk 8]
```diff
@@ -1,4 +1,30 @@
-﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
-// for details on configuring this project to bundle and minify static web assets.
-
-// Write your JavaScript code.
+﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/wwwroot/js/site.js:1]: Remove the comments

Comment (nicolasgranata) [reply src/DonateTo.IdentityServer/wwwroot/js/site.js:1]: Change the name of the site.js to "signup.js"

File: src/DonateTo.IdentityServer/wwwroot/js/site.js [hunk 9]
```diff
@@ -1,4 +1,30 @@
...
-// for details on configuring this project to bundle and minify static web assets.
-
-// Write your JavaScript code.
+﻿// Please see documentation at https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification
+// for details on configuring this project to bundle and minify static web assets.
+
+// Write your JavaScript code.
+
+function CheckPasswordCriteria(passwordValue) {
+  const element = document.getElementById("password");
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/wwwroot/js/site.js:7]: change the name of the variable, maybe password?

File: src/DonateTo.IdentityServer/wwwroot/js/site.js [hunk 10]
```diff
@@ -1,4 +1,30 @@
...
+    element.classList.add("is-valid");
+  } else {
+    element.classList.remove("is-valid");
+    element.classList.add("is-invalid");
+  }
+}
+
+function CheckPasswordMatch(confirmPasswordValue) {
+  const passwordValue = document.getElementById("password").value;
+  const element = document.getElementById("confirmPassword");
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/wwwroot/js/site.js:21]: Change the name of the variable to maybe confirmPasswordLabel or confirmPasswordInput, whatever the element is

File: src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs [hunk 11]
```diff
@@ -2,28 +2,31 @@
...
 
-        public string UserName { get => Email; }
+        public string UserName => Email;
 
         public string IdentityNumber { get; set; }
 
         public string Phone { get; set; }
 
         [Required(ErrorMessage = "Password is required")]
+        [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).{8,}$", ErrorMessage = "Password entry does not meet criteria.")]
```
Comment (guillemurano) [inline src/DonateTo.IdentityServer/Models/UserRegistrationViewModel.cs:24]: Password requirements should be configured in appsettings IdentityServer configuration

File: src/DonateTo.IdentityServer/Controllers/AccountController.cs [hunk 12]
```diff
@@ -142,16 +142,16 @@ public IActionResult Register()
...
             {
-                return View(userModel);
+                return View(userRegistrationViewModel);
             }
 
-            var user = _mapper.Map<User>(userModel);
+            var user = _mapper.Map<User>(userRegistrationViewModel);
 
-            var result = await _userManager.CreateAsync(user, userModel.Password);
+            var result = await _userManager.CreateAsync(user, userRegistrationViewModel.Password);
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/Controllers/AccountController.cs:154]: Add ConfigureAwait(false)

File: src/DonateTo.IdentityServer/wwwroot/js/singup.js [hunk 13]
```diff
@@ -0,0 +1,25 @@
+﻿function CheckPasswordCriteria(passwordValue) {
```
Comment (nicolasgranata) [inline src/DonateTo.IdentityServer/wwwroot/js/singup.js:1]: the name of the js should signup

**Pull request merged**